analyzers/src/SonarAnalyzer.CSharp/Rules/MethodParameterUnused.cs

```diff
@@ -126,8 +126,8 @@ private void ReportUnusedParametersOnMethod(MethodContext declaration)
 
         private void ReportOnDeadParametersAtEntry(MethodContext declaration, IImmutableList<IParameterSymbol> noReportOnParameters)
         {
-            var bodyNode = (CSharpSyntaxNode)declaration.Body ?? declaration.ExpressionBody;
-            if (bodyNode == null || declaration.Context.Node.IsKind(SyntaxKind.ConstructorDeclaration))
+            var declarationNode = (CSharpSyntaxNode)declaration.MethodDeclaration ?? declaration.LocalFunctionDeclaration;
```

Comment:
Is the LocalFunctionDeclaration fixing old FNs? That would be nice to extract in a dedicated PR with issue

Reply:
No, I found my answer later. So the resolving was intentional.